src/Microsoft.TestPlatform.CrossPlatEngine/TestEngine.cs

```diff
@@ -131,7 +131,7 @@ public IProxyExecutionManager GetExecutionManager(IRequestData requestData, ITes
 
                 var requestSender = new TestRequestSender(requestData.ProtocolConfig, hostManager.GetTestHostConnectionInfo());
 
-                return isDataCollectorEnabled ? new ProxyExecutionManagerWithDataCollection(requestData, requestSender, hostManager, new ProxyDataCollectionManager(requestData, testRunCriteria.TestRunSettings))
+                return isDataCollectorEnabled ? new ProxyExecutionManagerWithDataCollection(requestData, requestSender, hostManager, new ProxyDataCollectionManager(requestData, testRunCriteria.TestRunSettings), testRunCriteria)
```

Comment:
Don't pass testRunCriteria to ProxyExecutionManagerWithDataCollection. Pass it directly to ProxyDataCollectionManager.
Replace testRunCriteria.TestRunSettings with testRunCriteria.

Reply:
Done.